src/lib.rs

```diff
@@ -581,9 +642,43 @@ impl Default for SymmetricKey {
     }
 }
 
+/// Nonce structure used for authenticated symmetric encryption.
+#[derive(Debug, PartialEq, Eq, Serialize, Deserialize, Clone)]
+pub struct Nonce {
+    nonce: secretbox::Nonce,
```

Comment:
feels strange to call it `nonce.nonce`. Since the inner type is not exposed (as we provide `impl Nonce` below to abstract it) it's only known to this module, so maybe `inner` is better ? Or `impl Deref` or something to get the inner so that we can use it as `*nonce` here (which returns `&rust_sodium::Nonce`). However pls see what the best practices are and do that - think `Deref` is not always non-controversial so might or might not be a good fit here.

Reply:
We already have `Signature.signature`, `SecretSignKey.sign`, etc. The use of `inner` seems to be reserved for when we have the inner value wrapped in a smart pointer (`Arc` in this crate). So I chose `Nonce.nonce` for consistency even if it looks a little strange.